torch/csrc/jit/codegen/cuda/lower_unroll.cpp

```diff
@@ -105,10 +91,8 @@ void UnrollPass::handle(kir::Expr* expr) {
     // For expr calling a device func with block sync, don't create
     // if-then-else but pass the predicate to the device func
     if (ir_utils::hasBlockSync(expr, GpuLower::current()->threadPredMap())) {
-      // All threads should join blockBroadcast
-      auto thread_pred = expr->isA<kir::BroadcastOp>()
-          ? ir_builder.trueVal()
-          : GpuLower::current()->threadPredMap().getExpr(out_tv->fuserTv());
```

Comment:
This special handling of masking the thread predicate when the expr is `blockBroadcast` is now encapsulated in `ThreadPredicateMap::getPredicate`.